src/SExpr.purs

```diff
@@ -256,17 +256,14 @@ paragraphElemsFwd
    => MonadError Error m
    => List (ParagraphElem a)
    -> m (E.Expr a)
-paragraphElemsFwd =
-   foldr step (pure (enil bot))
-   where
-   step :: ParagraphElem a -> m (E.Expr a) -> m (E.Expr a)
-   step (Token s) accM = do
-      acc <- accM
-      pure (econs bot (E.Constr bot cText (E.Str bot s : Nil)) acc)
-   step (Unquote e) accM = do
-      acc <- accM
-      e' <- desug e
-      pure (econs bot (E.Constr bot cText (e' : Nil)) acc)
+paragraphElemsFwd Nil = pure (enil bot)
+paragraphElemsFwd (Token s : elems) = do
+   elems' <- paragraphElemsFwd elems
```

Comment:
Think about the type of the variable -- that should determine the name. Here `elems'` has type `E.Expr a` (because its produced by a call to `paragraphElemsFwd`. So the convention should be `e` here.